exporters/metric/prometheus/prometheus.go

```
@@ -126,59 +138,60 @@ func NewRawExporter(config Config) (*Exporter, error) {
 // InstallNewPipeline instantiates a NewExportPipeline and registers it globally.
 // Typically called as:
 //
-// 	pipeline, hf, err := prometheus.InstallNewPipeline(prometheus.Config{...})
+// 	hf, err := prometheus.InstallNewPipeline(prometheus.Config{...})
 //
 // 	if err != nil {
 // 		...
 // 	}
 // 	http.HandleFunc("/metrics", hf)
 // 	defer pipeline.Stop()
 // 	... Done
-func InstallNewPipeline(config Config, options ...push.Option) (*push.Controller, *Exporter, error) {
-	controller, exp, err := NewExportPipeline(config, options...)
+func InstallNewPipeline(config Config, options ...pull.Option) (*Exporter, error) {
+	exp, err := NewExportPipeline(config, options...)
 	if err != nil {
-		return controller, exp, err
+		return nil, err
 	}
-	global.SetMeterProvider(controller.Provider())
-	return controller, exp, err
+	global.SetMeterProvider(exp.Provider())
+	return exp, nil
 }
 
-// NewExportPipeline sets up a complete export pipeline with the recommended setup,
-// chaining a NewRawExporter into the recommended selectors and integrators.
-//
-// The returned Controller contains an implementation of
-// `metric.Provider`.  The controller is returned unstarted and should
-// be started by the caller to begin collection.
-func NewExportPipeline(config Config, options ...push.Option) (*push.Controller, *Exporter, error) {
-	exporter, err := NewRawExporter(config)
-	if err != nil {
-		return nil, nil, err
-	}
-
-	// Prometheus uses a stateful push controller since instruments are
+// SetController sets up a standard *pull.Controller as the metric provider
+// for this exporter.
+func (e *Exporter) SetController(config Config, options ...pull.Option) error {
```

Does this need to return an error?

---

No, fixed.